### ISSUE TITLE
[Bug] Updating Extensions Hangs, no Trust Prompt presented to user until restart.

### ISSUE THREAD

**Issue by jedinjapan** [Jays2Kings/tachiyomiJ2K#1734]
### Steps to reproduce

1. User has extensions installed.
2. Extension detects an update from a new repo.
3. Update > Install > nothing.

App must be closed and reopened for the "Trust" prompt to appear when you scroll through them on sources

### Expected behavior

Update > Install > Trust Check > Trust Prompt > Install Finished > Extension is useable

If there is an issue with repo changing causing a trust issue, the app should prompt the user after install immediately to trust, or assume trust from the install prompt being approved manually. Alternatively, trust could be managed at a per-repo level, and updates from the same source not incur this check.

### Actual behavior

Update > Install > Install Completes > Nothing

Extension still appears prompting to update, user can begin a second install which will hang, but cannot proceed to trust the extension until restarting the app, where any pending trust prompts will show all at once.

### TachiyomiJ2K version

1.7.4

### Android version

Android 14

### Device

Google Pixel 7

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with an extension, I should be opening an issue in the [extensions repository](https://github.com/tachiyomiorg/tachiyomi-extensions/issues/new/choose).
- [X] I am reporting an issue exclusive to this fork. I have also checked that is not an issue on the [main version of Tachiyomi](https://github.com/tachiyomiorg/tachiyomi)
- [X] I have tried the [troubleshooting guide](https://tachiyomi.org/help/guides/troubleshooting/).
- [X] I have updated the app to version **[1.7.4](https://github.com/Jays2Kings/tachiyomiJ2K/releases/latest)**.
- [X] I have updated all installed extensions.
- [ ] I have admitted that I am a clown by having checked this box, as I have not read these acknowledgements.
- [X] I have filled out all of the requested information in this form.

**Comment by AdamHavlicek** [comment]
I came up with this fix and it might be a solution for this bug. 

https://github.com/Jays2Kings/tachiyomiJ2K/blob/8946395f78d22e44471497d3b76de661e9b4166e/app/src/main/java/eu/kanade/tachiyomi/extension/ExtensionManager.kt#L410

```kotlin
  override fun onExtensionUntrusted(extension: Extension.Untrusted) {
            val installedExtension = _installedExtensionsFlow.value
                .find { it.pkgName == extension.pkgName }
            if (installedExtension != null) {
                _installedExtensionsFlow.value -= installedExtension
            }
            _untrustedExtensionsFlow.value += extension
        }

```
